components/src/directives/focus-first-element/index.ts

```diff
@@ -0,0 +1,38 @@
+import {Directive} from 'vue';
+
+export interface FocusFirstHTMLElement extends HTMLElement {
+  activeElement?: Element | null;
+}
+
+const focusableElements = 'input, select, textarea, [tabindex], [href]';
+const excludeCloseButton = 'button:not(.oxd-dialog-close-button)';
+
+const focusOnFirstElement = (element: Element, matchingString: string) => {
+  const firstFocusableElement = element.querySelectorAll(matchingString)[0];
+  if (firstFocusableElement) {
+    (firstFocusableElement as HTMLElement).focus();
+  }
+};
+const focusonFirstElementDirective: Directive = {
+  mounted(el: FocusFirstHTMLElement) {
+    el.activeElement = document.activeElement;
+    focusOnFirstElement(el, focusableElements + ', ' + excludeCloseButton);
+  },
+  beforeUnmount(el: FocusFirstHTMLElement) {
+    if (el.activeElement && (el.activeElement as HTMLElement).offsetParent) {
```

Comment:
not sure what this achieve? activeElement is already in focus?

Reply:
This is when the previous active element before the modal popup needs to be focused after the modal closes but that element is disappeared. Ex. The report delete button only appears when hovering on the row, but after closing the modal, if we try to focus on it, it's not there since we are not hovering over it.